C client header - need C & C++ compatibility

Suggestion: add `extern "C"` on `tb_client.h`.

Given error:
```bash
[build] [ 80%] Linking CXX executable tb_cpp
[build] ld.lld: error: undefined symbol: tb_client_submit(void*, tb_packet_list_t*)
[build] >>> referenced by tb_client.hpp:106 (/home/kassane/Documents/tb-client-cpp/include/tb_client.hpp:106)
[build] >>>               CMakeFiles/tb_cpp.dir/src/main.cpp.o:(tb::Client::send_request(void*, tb_packet_list_t*, tb::CompletionContext*))
[build] >>> did you mean: extern "C" tb_client_submit
[build] >>> defined in: /home/kassane/Documents/tb-client-cpp/vendor/lib/libtb_client.so
[build] make[2]: *** [CMakeFiles/tb_cpp.dir/build.make:99: tb_cpp] Error 1
[build] make[1]: *** [CMakeFiles/Makefile2:99: CMakeFiles/tb_cpp.dir/all] Error 2
[build] make: *** [Makefile:136: all] Error 2
[proc] The command: /usr/bin/cmake --build /home/kassane/Documents/tb-client-cpp/build --config Debug --target all -j 18 -- exited with code: 2
[driver] Build completed: 00:00:00.167
[build] Build finished with exit code 2
```

--- a/src/clients/c/tb_client.h
+++ b/src/clients/c/tb_client.h
@@ -5,6 +5,10 @@
 
 #ifndef TB_CLIENT_H
 #define TB_CLIENT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
  
 #include <stddef.h>
 #include <stdint.h>
@@ -219,5 +223,8 @@ void tb_client_deinit(
     tb_client_t client
 );
 
-#endif // TB_CLIENT_H
+#ifdef __cplusplus
+} // extern "C"
+#endif
 
+#endif // TB_CLIENT_H

--- a/src/clients/c/tb_client_header.zig
+++ b/src/clients/c/tb_client_header.zig
@@ -143,6 +143,10 @@ pub fn main() !void {
         \\
         \\#ifndef TB_CLIENT_H
         \\#define TB_CLIENT_H
+        \\
+        \\#ifdef __cplusplus
+        \\extern "C" {{
+        \\#endif
         \\ 
         \\#include <stddef.h>
         \\#include <stdint.h>
@@ -217,6 +221,13 @@ pub fn main() !void {
         \\
     , .{});
 
-    try buffer.writer().print("#endif // TB_CLIENT_H\n\n", .{});
+    try buffer.writer().print(
+        \\#ifdef __cplusplus
+        \\}} // extern "C"
+        \\#endif
+        \\
+        \\#endif // TB_CLIENT_H
+        \\
+    , .{});
     try std.fs.cwd().writeFile("src/clients/c/tb_client.h", buffer.items);
 }

--- a/src/clients/go/pkg/native/tb_client.h
+++ b/src/clients/go/pkg/native/tb_client.h
@@ -5,6 +5,10 @@
 
 #ifndef TB_CLIENT_H
 #define TB_CLIENT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
  
 #include <stddef.h>
 #include <stdint.h>
@@ -219,5 +223,8 @@ void tb_client_deinit(
     tb_client_t client
 );
 
-#endif // TB_CLIENT_H
+#ifdef __cplusplus
+} // extern "C"
+#endif
 
+#endif // TB_CLIENT_H

